firestore.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict, List, Optional
```

Comment:
Would be nice to have this documented more clearly.  At least explaining what you're doing at a high level and why you think this is the right approach is important.

I'm not sure why this is done in an OO manner, for example, or where / when a job object should be updated.